### PR TITLE
chore: deposit and withdrawal flow

### PR DESCRIPTION
**Pull request by dangelovska**

### CODE REVIEW

File: rockwallet/rockwallet/Helpers/ApplicationController.swift [hunk 1]
```diff
@@ -470,8 +469,13 @@ class ApplicationController: Subscriber {
             self?.coordinator?.showProfile()
         }
         
-        homeScreen.didTapProSegment = { [weak self] in
-            self?.coordinator?.showComingSoon(reason: .rockWalletPro, restrictionReason: .state)
+        homeScreen.didTapProSegment = { [weak self] isUserLogged in
+            guard let isUserLogged = isUserLogged else { return }
```
Comment (dgacevic) [inline rockwallet/rockwallet/Helpers/ApplicationController.swift:473]: `guard let isUserLogged else { return }`

File: rockwallet/rockwallet/ProBalancesWorker.swift [hunk 2]
```diff
@@ -17,14 +17,25 @@ struct ProBalancesRequestData: RequestModelData {
...
+    var bsv: Decimal?
 }
 
 struct ProBalancesModel: Model {
+    var mnet: Decimal?
+    var usdc: Decimal?
+    var btc: Decimal?
+    var eth: Decimal?
+    var bsv: Decimal?
 }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/ProBalancesWorker.swift:33]: these can be non-optional with default value of 0?

File: rockwallet/rockwallet/Helpers/ApplicationController.swift [hunk 3]
```diff
@@ -470,8 +469,13 @@ class ApplicationController: Subscriber {
...
         
-        homeScreen.didTapProSegment = { [weak self] in
-            self?.coordinator?.showComingSoon(reason: .rockWalletPro, restrictionReason: .state)
+        homeScreen.didTapProSegment = { [weak self] isUserLogged in
+            guard let isUserLogged else { return }
+            if isUserLogged {
+                self?.coordinator?.showComingSoon(reason: .rockWalletPro, restrictionReason: .state)
+            } else {
+                self?.coordinator?.showProOption()
+            }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/Helpers/ApplicationController.swift:478]: either keep guard or if

File: rockwallet/rockwallet/ProBalancesWorker.swift [hunk 4]
```diff
@@ -17,14 +17,25 @@ struct ProBalancesRequestData: RequestModelData {
...
 
 class ProBalancesMapper: ModelMapper<ProBalancesResponseData, ProBalancesModel> {
     override func getModel(from response: ProBalancesResponseData?) -> ProBalancesModel? {
-        return .init()
+        guard let response = response else { return nil }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/ProBalancesWorker.swift:37]: can be removed

File: rockwallet/rockwallet/src/Redesign/Core/CLEANerSwift/Cycles/CreateTransaction/CreateTransactionVIP.swift [hunk 5]
```diff
@@ -165,6 +146,7 @@ extension Interactor where Self: CreateTransactionViewActions,
         }
         
         completion?(error)
+        return
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/Redesign/Core/CLEANerSwift/Cycles/CreateTransaction/CreateTransactionVIP.swift:149]: why?

File: rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/TransferFundsInteractor.swift [hunk 6]
```diff
@@ -108,8 +114,53 @@ class TransferFundsInteractor: NSObject, Interactor, TransferFundsViewActions {
...
+            }
+        }
+    }
+    
+    func prepareFees(viewAction: AssetModels.Fee.ViewAction, completion: (() -> Void)?) {
+        guard let from = dataStore?.fromAmount else {
+            return
+        }
+        
+        generateSender(viewAction: .init(fromAmountCurrency: dataStore?.fromAmount?.currency))
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/TransferFundsInteractor.swift:159]: you can use "from" on line 155

File: rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/WithdrawalWorker.swift [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+import Foundation
+
+struct WithdrawalRequestData: RequestModelData {
+    let amount: String?
+    let address: String?
+    let asset: String?
+    
+    func getParameters() -> [String: Any] {
+        let params = [
+            "amount": amount,
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/WithdrawalWorker.swift:20]: can be moved to the line above

File: rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/WithdrawalWorker.swift [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            "address": address,
+            "asset": asset
+        ]
+        return params.compactMapValues { $0 }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/Redesign/Scenes/TransferFunds/WithdrawalWorker.swift:24]: empty line above for consistency, please

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 9]
```diff
@@ -407,6 +413,21 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
...
         
+        if selectedSegment == .rockWalletPro {
+            ProBalancesWorker().execute(requestData: ProBalancesRequestData()) { result in
+                switch result {
+                case .success(let data):
+                    self.updateProBalance(data: data)
+                    self.assetListTableView.proBalancesData = data
+                    
+                case .failure:
+                    break
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:424]: no error displaying?

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 10]
```diff
@@ -550,6 +572,14 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
         totalAssetsAmountLabel.text = String(format: "%@ %@", formattedBalance, fiatCurrency)
     }
     
+    func updateProBalance(data: ProBalancesModel?) {
+        let balance = 0.0
+        
+        guard let formattedBalance = ExchangeFormatter.fiat.string(for: balance),
+              let fiatCurrency = Store.state.orderedWallets.first?.currentRate?.code else { return }
+        totalAssetsAmountLabel.text = String(format: "%@ %@", formattedBalance, fiatCurrency)
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:580]: Constant.currencyFormat

File: rockwallet/rockwallet/src/Views/HomeScreenCell.swift [hunk 11]
```diff
@@ -72,32 +74,53 @@ class HomeScreenCell: UITableViewCell, Subscriber {
...
+            case Constant.USDC:
+                proBalance = proBalancesData?.usdc ?? 0
+            default:
+                break
+            }
+        }
+        
+        guard let formattedBalance = ExchangeFormatter.fiat.string(for: proBalance),
+              let fiatCurrency = Store.state.orderedWallets.first?.currentRate?.code else { return }
+        let proBalanceText = String(format: "%@ %@", formattedBalance, fiatCurrency)
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/Views/HomeScreenCell.swift:103]: Constant.currencyFormat

File: rockwallet/rockwallet/ProBalancesWorker.swift (hunk 2 above)
Comment (dangelovska) [reply rockwallet/rockwallet/ProBalancesWorker.swift:33]: if we don't have any balance on some of the assets, BE is not returning it.

File: rockwallet/rockwallet/Helpers/ApplicationController.swift [hunk 12]
```diff
@@ -470,7 +469,13 @@ class ApplicationController: Subscriber {
             self?.coordinator?.showProfile()
         }
         
-        homeScreen.didTapProSegment = { [weak self] in
+        homeScreen.didTapProSegment = { [weak self] isUserLogged in
+            guard let isUserLogged else { return }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/Helpers/ApplicationController.swift:473]: you don't need this

File: rockwallet/rockwallet/ProBalancesWorker.swift (hunk 2 above)
Comment (kenanmamedoff) [reply rockwallet/rockwallet/ProBalancesWorker.swift:33]: unrelated. this is not the response model but map model. you can make those values non optional and assign 0 in the mapper.

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**